phoenix-core/src/main/java/org/apache/phoenix/parse/ColumnDef.java

```diff
@@ -78,7 +79,7 @@
          }
          
          this.isNull = isNull;
-         if (this.dataType == PChar.INSTANCE) {
+         if (this.dataType == PChar.INSTANCE || this.dataType == PCharArray.INSTANCE) {
```

Comment:
I'd be curious if BINARY type would require the same fix. A more general check would be:
    if (this.dataType.isFixedWidth()) {

along with generalizing the error messages.